### ISSUE TITLE
[icetray] pybindings for map_string_I3Configuration (Trac #1531)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1523]
It's possible to get an `icecube.icetray.map_string_I3Configuration` object in python, so let's make sure it's actually usable instead of an opaque object.

Migrated from https://code.icecube.wisc.edu/ticket/1531
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:53", 
    "description": "It's possible to get an `icecube.icetray.map_string_I3Configuration` object in python, so let's make sure it's actually usable instead of an opaque object.", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067233566919", 
    "component": "combo core", 
    "summary": "[icetray] pybindings for map_string_I3Configuration", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-01-27T17:34:43", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160328T17:56:08: 
* olivas changed status from "new" to "accepted"
* olivas changed owner from "" to "olivas"

**Comment by icecube-trac** [comment]
Trac update at 20160420T18:52:16: olivas changed milestone from "March 2016" to "April 2016"

**Comment by icecube-trac** [comment]
Trac update at 20160814T23:49:54: 
* olivas changed status from "accepted" to "closed"
* olivas changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20160815T01:39:55: 
* olivas changed status from "closed" to "reopened"
* olivas changed resolution from "fixed" to ""

**Comment by icecube-trac** [comment]
Trac update at 20170220T18:36:25: olivas changed status from "reopened" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20170330T18:49:55: olivas changed milestone from "May 2016" to "May 2017"

**Comment by icecube-trac** [comment]
Trac update at 20170524T16:48:50: 
* david.schultz changed owner from "olivas" to "david.schultz"
* david.schultz changed status from "accepted" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20170524T17:10:06: david.schultz commented:

> This one is nasty.  We expose `__len__`, `__contains__`, and `__iter__`, so you can iterate over it, but good luck finding keys, values, etc.  There are no actual properties or methods available.
> 
> Going to add them now.

**Comment by icecube-trac** [comment]
Trac update at 20170524T22:24:23: 
* david.schultz changed status from "assigned" to "closed"
* david.schultz commented:

> In changeset 2901:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2901"
> fill in the python bindings for I3Configuration. remove the description proxy in favor of just returning a dict, since it's simple and easy. fixes #1523
> ```


* david.schultz changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190111T20:59:03: nwhitehorn commented:

> In changeset 168817:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="168817"
> fill in the python bindings for I3Configuration. remove the description proxy in favor of just returning a dict, since it's simple and easy. fixes #1523
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:53: 
* olivas changed milestone from "May 2017" to ""
* olivas commented:

> Milestone May 2017 deleted
